src/Symfony/Component/DependencyInjection/Compiler/ResolveReferencesToAliasesPass.php

```diff
@@ -34,7 +34,10 @@ public function process(ContainerBuilder $container)
             $this->currentId = $id;
 
             if ($aliasId !== $defId = $this->getDefinitionId($aliasId, $container)) {
-                $container->setAlias($id, $defId)->setPublic($alias->isPublic())->setPrivate($alias->isPrivate());
+                $container->setAlias($id, $defId)
+                    ->setPublic($alias->isPublic())
+                    ->setPrivate($alias->isPrivate())
+                    ->setDeprecated($alias->isDeprecated(), $alias->getDeprecationTemplate());
```

Comment:
why should a deprecated alias propagate to its referring aliases?

Reply:
I'll reset that to the previous code